ow_faults_detection/src/FaultDetector.cpp

```diff
@@ -0,0 +1,256 @@
+  string ant_tilt_str = "/ant_tilt_position_controller";
+  // m_ant_pan_command_sub = node_handle.subscribe( ant_pan_str + string("/command"),
+  //                                         10,
+  //                                         &FaultDetector::antennaPanCommandCb,
+  //                                         this);
+  // m_ant_pan_state_sub = node_handle.subscribe(ant_pan_str + string("/state"),
+  //                                         10,
+  //                                         &FaultDetector::antennaPanStateCb,
+  //                                         this);
+  // m_ant_tilt_command_sub = node_handle.subscribe( ant_tilt_str + string("/command"),
+  //                                         10,
+  //                                         &FaultDetector::antennaTiltCommandCb,
+  //                                         this);
+  // m_ant_tilt_state_sub = node_handle.subscribe(ant_tilt_str + string("/state"),
+  //                                         10,
+  //                                         &FaultDetector::antennaTiltStateCb,
+  //                                         this);
+
+  // camera
+
+  const char* image_str = "/StereoCamera/left/image_";
+  m_camera_original_trigger_sub = node_handle.subscribe( image_str + string("trigger"),
+    10, &FaultDetector::camerTriggerCb, this);
+  m_camera_raw_sub = node_handle.subscribe(image_str + string("raw"),
+    10, &FaultDetector::cameraRawCb, this);
+  
+  m_camera_trigger_timer = node_handle.createTimer(ros::Duration(0.1), &FaultDetector::cameraTriggerPublishCb, this);
+
+  //  power fault publishers and subs
+  m_power_soc_sub = node_handle.subscribe("/power_system_node/state_of_charge",
+                                          10,
+                                          &FaultDetector::powerSOCListener,
+                                          this);
+  m_power_temperature_sub = node_handle.subscribe("/power_system_node/battery_temperature",
+                                                  10,
+                                                  &FaultDetector::powerTemperatureListener,
+                                                  this);
+
+  // topics for JPL msgs: system fault messages, see Faults.msg, Arm.msg, Power.msg, PTFaults.msg
+  m_arm_fault_msg_pub = node_handle.advertise<ow_faults_detection::ArmFaults>("/faults/arm_faults_status", 10);
+  m_antenna_fault_msg_pub = node_handle.advertise<ow_faults_detection::PTFaults>("/faults/pt_faults_status", 10);
+  m_camera_fault_msg_pub = node_handle.advertise<ow_faults_detection::CamFaults>("/faults/cam_faults_status", 10);
+  m_power_fault_msg_pub = node_handle.advertise<ow_faults_detection::PowerFaults>("/faults/power_faults_status", 10);
+  m_system_fault_msg_pub = node_handle.advertise<ow_faults_detection::SystemFaults>("/faults/system_faults_status", 10);
+
+}
+
+// Creating Fault Messages
+template<typename fault_msg>
+void FaultDetector::setFaultsMessageHeader(fault_msg& msg){
```

Comment:
Throughout your code, be consistent on whether a function's opening brace is on the first line, or a separate line.  I prefer the latter, and that's what most OW code uses, and what you use in most places.